apps/04TowersOfHanoi.js

```diff
@@ -1,93 +1,190 @@
-  console.log("b: " + stacks.b);
-  console.log("c: " + stacks.c);
+    console.log("a: " + stacks.a);
+    console.log("b: " + stacks.b);
+    console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+var moveCount = 0;
+
+function movePiece(startStack, endStack) {
+    // Your code here
+    //get the piece off of the top of the start stack and move to the end stack and save it.
+    if (isLegal(startStack, endStack)) {
+        stacks[endStack].push(stacks[startStack].pop());
+    }
 
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
+    // Your code here
+    //get a reference to the actual startStack and save it (putting something into a variable)
+    //does start stack have anything in it??
+    //get a reference to the actual endStack and save it
+    //is the piece from the start stack bigger or smaller than the piece on top of the end stack? legal move would //only be smaller on bigger.
+    //does the end stack have anything it??
+    //if bad return something, if good return something else
+    var startStackValue = stacks[startStack][stacks[startStack].lenth - 1];
+
+    if (stacks[startStack].length === 0) {
+        var startStackValue = 0;
+    } else {
+        var startStackValue = stacks[startStack][stacks[startStack].length - 1];
+    }
+
+    if (stacks[endStack].length === 0) {
+        var endStackValue = 0;
+    } else {
+        var endStackValue = stacks[endStack][stacks[endStack].length - 1];
+    }
+
+    if (startStackValue < endStackValue || endStackValue == 0) {
+        return true;
+    } else {
+        console.log("Illegal Move");
+        return false;
+    }
 
 }
 
 function checkForWin() {
-  // Your code here
+    // Your code here
+    //does stack c have a .length of 4?
+    console.log("Stacks a is " + stacks.a.length);
+    console.log("Stacks a is " + stacks.b.length);
+    console.log("Stacks a is " + stacks.c.length);
+
+    if (stacks.b.length == 4 || stacks.c.length == 4) {
+        console.log("You win, your movecount was" + moveCount);
+        return true;
+        processExit();
+    } else {
+        return false;
+    }
+}
+
 
+function checkEntry(startStack, endStack) {
+    if (startStack == 'a' || startStack == 'b' || startStack == 'c') {
+        return true;
+    } else {
+        console.log("Please enter 'a' , 'b' , or 'c'");
+        return false;
+    }
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
 
+function towersOfHanoi(startStack, endStack) {
+    // Your code here: coordinates the effort of all the other functions
+    //check if the move is okay. if it is, make the move.
+    //check for win after move is made
+    startStack = startStack.toLowerCase();
+    endStack = endStack.toLowerCase();
+
+    if (startStack == "win") {
+        winTheGame();
+    }
+
+    if (checkEntry(startStack, endStack) && isLegal(startStack, endStack)) {
+        movePiece(startStack, endStack);
```

Comment:
With you calling isLegal() inside the movePiece() function, it looks like you call the isLegal twice here..